Odata-docs/client/v6/deal-with-server-driven-paging.md

```diff
@@ -19,15 +19,127 @@ Example:
 var context = new DefaultContainer(new Uri("https://services.odata.org/v4/TripPinServiceRW/"));
 
 // DataServiceQueryContinuation<T> contains the next link
-DataServiceQueryContinuation<Person> token = null;
+DataServiceQueryContinuation<Person> nextLink = null;
 
 // Get the first page
 var response = await context.People.ExecuteAsync() as QueryOperationResponse<Person>;
 
 // Loop if there is a next link
-while ((token = response.GetContinuation()) != null)
+while ((nextLink = response.GetContinuation()) != null)
 {
     // Get the next page
-    response = await context.ExecuteAsync<Person>(token) as QueryOperationResponse<Person>;
+    response = context.Execute<Person>(nextLink);
 }
 ```
+
+## Nested Pagination
+There are instances where we need to load pages of entities as well as pages of their related entities.
+The example below returns related Trips for each Person entity from the data service. It uses a do...while loop to paginate through Person entities and a nested while loop to paginate through the related Trips.
+We need to enumerate the response before calling `GetContinuation` on it.
+
+``` csharp
+var context = new DefaultContainer(new Uri("https://services.odata.org/v4/TripPinServiceRW/"));
+var pageCount = 0;
+var innerPageCount = 0;
```

Comment:
Shouldn't the `innerPageCount` variable be initialized at 0 for each person?

Reply:
#Resolved